tools/gce-google-keys-to-cmek/main.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+
+# Copyright 2019 Google, LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import googleapiclient
+import googleapiclient.discovery
+import time
+import re
+import sys
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert disks attached to a GCE instance from Google-managed encryption keys to customer-managed encryption keys.')
+    parser.add_argument('--project', required=True, dest='project', action='store', type=str, help='Project containing the GCE instance.')
+    parser.add_argument('--zone', required=True, dest='zone', action='store', type=str, help='Zone containing the GCE instance.')
+    parser.add_argument('--instance', required=True, dest='instance', action='store', type=str, help='Instance name.')
+    parser.add_argument('--key-ring', required=True, dest='keyRing', action='store', type=str, help='Name of the key ring containing the key to encrypt the disks. Must be in the same zone as the instance.')
+    parser.add_argument('--key-name', required=True, dest='keyName', action='store', type=str, help='Name of the key to encrypt the disks. Must be in the same zone as the instance.')
+    parser.add_argument('--key-version', required=True, dest='keyVersion', action='store', type=int, help='Version of the key to encrypt the disks.')
+    parser.add_argument('--destructive', dest='destructive', action='store_const', const=True, default=False, help='Upon completion, delete source disks and snapshots created during migration process.')
+    args = parser.parse_args()
+
+    migrate_instance_to_cmek(args.project, args.zone, args.instance, args.keyRing, args.keyName, args.keyVersion, args.destructive)
+
+def migrate_instance_to_cmek(project, zone, instance, keyRing, keyName, keyVersion, destructive):
+    start = time.time()
+
+    zone_regexp = r'^(\w\w-\w*\d)-(\w)$'
+    region = re.search(zone_regexp, zone).group(1)
+
+    compute = googleapiclient.discovery.build('compute', 'v1')
+
+    # TODO: Consider if we can use compute.disks().list() and do disk checking
```

Comment:
Do you mean to implement this before merging?

Reply:
No, its just an alternative implementation suggestion